spharpy/samplings/coordinates.py

```diff
@@ -310,3 +310,22 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    def _set_points(self, x, y, z):
+        # set points first to make sure data is valid
+        super()._set_points(x, y, z)
+
+        # check if all points have the same radius
+        radius = np.sqrt(
+            self.x.flatten()**2 +
+            self.y.flatten()**2 +
+            self.z.flatten()**2)
```

Comment:
```suggestion
        radius = self.radius
```
since it is already set here self.radius should work as expected.

Reply:
Thanks for catching. In the initial version points were set after testing for the radius but this is more elegant and less prone to errors.